cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -3360,3 +3477,144 @@ def _migrate_to_fcd(self, context, volume, host):
             nova_api.update_server_volume(context, instance_uuid,
                                           volume.id, volume.id)
         return (True, None)
+
+    def _start_periodic_tasks(self):
+        if self.configuration.enable_image_cache:
+            lock_name = f"{self._cache_project_name()}-purge_cache"
+
+            @coordination.synchronized(lock_name)
+            def _purge_image_cache_locked():
+                self._purge_image_cache()
+
+            LOG.info("Starting the purge_image_cache periodic task.")
+            image_cache_ptask = loopingcall.FixedIntervalLoopingCall(
+                _purge_image_cache_locked,)
+            image_cache_ptask.start(
+                interval=self.configuration.image_cache_purge_interval,
+                stop_on_exception=False)
+
+    def _purge_image_cache(self):
+        """Deletes expired images from the cache"""
+        cached_images = self._get_cached_images()
+        age_seconds = self.configuration.image_cache_age_seconds
+        for cached_image in cached_images:
+            if not timeutils.is_older_than(
+                    cached_image['created_at'],
+                    age_seconds):
+                continue
+
+            try:
+                self.volumeops.delete_backing(cached_image['vm_ref'])
+                LOG.debug("Deleted expired image (%s) from the cache.",
+                          cached_image['name'])
+            except exceptions.VimException:
+                LOG.exception("Failed to delete the expired image %s",
+                              cached_image['name'])
+
+    def _get_cached_images(self):
```

Comment:
We run this function every minute with the stats generation (I think). How long does it take to run in a real-world environment? Do we have to optimize somewhere?